src/main/java/edu/ucsb/nceas/mdqengine/scheduler/MonitorJob.java

```diff
@@ -0,0 +1,300 @@
+            if (DataONEauthToken == null) {
+                DataONEauthToken = cfg.getString("DataONE.authToken");
+                log.debug("Got token from properties file.");
+            } else {
+                log.debug("Got token from env.");
+            }
+        } catch (ConfigurationException | IOException ce) {
+            JobExecutionException jee = new JobExecutionException("error executing task.");
+            jee.initCause(ce);
+            throw jee;
+        }
+        Session session = DataONE.getSession(subjectId, DataONEauthToken);
+        return session;
+    }
+
+    /**
+     * Retrieves the metadata of an object specified by the provided pid
+     * from a specified node using the given session.
+     * 
+     * @param pidStr  the pid of the object
+     * @param nodeId  the identifier of the node where the object is located
+     * @param session the session used to authenticate the request
+     * @return an InputStream containing the metadata of the specified object
+     * 
+     * @throws JobExecutionException if there is an error executing the quartz task
+     */
+    public InputStream getMetadata(String pidStr, String nodeId, Session session) throws JobExecutionException {
+
+        String nodeServiceUrl = null;
+        MultipartRestClient mrc = null;
+        MultipartMNode mnNode = null;
+        MultipartCNode cnNode = null;
+        InputStream objectIS = null;
+        Identifier pid = new Identifier();
+        pid.setValue(pidStr);
+        log.info(pid.getValue());
+
+        try {
+            MDQconfig cfg = new MDQconfig();
+            String nodeAbbr = nodeId.replace("urn:node:", "");
+            nodeServiceUrl = cfg.getString(nodeAbbr + ".serviceUrl");
+
+        } catch (ConfigurationException | IOException ce) {
+            JobExecutionException jee = new JobExecutionException("Monitor: error executing task.");
+            jee.initCause(ce);
+            throw jee;
+        }
+
+        try {
+            mrc = new HttpMultipartRestClient();
+        } catch (Exception e) {
+            log.error("Monitor: error creating rest client: " + e.getMessage());
+            JobExecutionException jee = new JobExecutionException(e);
+            jee.setRefireImmediately(false);
+            throw jee;
+        }
+        log.info(nodeServiceUrl);
+        Boolean isCN = DataONE.isCN(nodeServiceUrl);
+        if (isCN) {
+            cnNode = new MultipartCNode(mrc, nodeServiceUrl, session);
+        } else {
+            mnNode = new MultipartMNode(mrc, nodeServiceUrl, session);
+        }
+
+        try {
+            if (isCN) {
+                objectIS = cnNode.get(session, pid);
+            } else {
+                objectIS = mnNode.get(session, pid);
+            }
+            log.trace("Monitor: Retrieved metadata object for pid: " + pidStr);
+        } catch (NotAuthorized na) {
+            log.error("Monitor: Not authorized to read pid: " + pidStr + ", unable to retrieve object");
```

Comment:
@mbjones can you have a look at the new code and let me know what you think?

Reply:
I took a look, looks good. Some minor points:

- probably use `log.debug()` rather than log.trace() on line 246
- NotAuthorized, InvalidToken, NotFound could just be transient issues, so I would log it as a warning and try again in the trycounter loop
- ServiceFailure and InsufficientResources are likely system problems so, log these as an error, but still we;'d want to save the task and reschedule it for later. There needs to be a way to let a metadig admin know about these so they can go add disk, fix system problems, etc.
- There are blank exception handling blocks on line 353 and line 363